homeassistant/components/tado/helper.py

```diff
@@ -49,3 +49,15 @@ def decide_duration(
         )
 
     return duration
+
+
+def generate_supported_fanmodes(tado_to_ha_mapping: dict[str, str], options: list[str]):
+    """Return correct list of fan modes or None."""
+    supported_fanmodes = [
+        tado_to_ha_mapping.get(option)
+        for option in options
+        if tado_to_ha_mapping.get(option) is not None
+    ]
+    if not supported_fanmodes:
```

Comment:
Put this above, to keep the early-return principle. It saves some unnecessary processing done above, if its None.

Reply:
I don't think I can put it above as this if statement is to return a None if I get a list[None] .
The main goal of this function is to avoid duplicate code between fanLevel and fanSpeed and return the good type for supported fan modes that should be a list[str] | None.

If i don't keep this if statement here, I will return list[str | None].

Reply:
I see, makes sense. Thanks for the quick fix, let's see if we can get this in Beta!